### ISSUE TITLE
ruff format: whitespace before `:` (PEP 8: E203)

### ISSUE THREAD

**Issue by Zerotask** [astral-sh/ruff#10196]
```python
field_types_1 = field_types[len(field_types) // 2 :]
``` 

PyCharm marks it as: `PEP 8: E203 whitespace before ':'`.
So the expected code looks like:
```python
field_types_1 = field_types[len(field_types) // 2:]
``` 

If I run ruff format / auto-format from the ruff plugin, it adds the whitespace again.

---

ruff 0.3.0 with preview enabled

**Comment by 57an** [comment]
This is black behaviour. You can add E203 in Ignored Errors list for PEP 8 coding style violetion check.

**Comment by charliermarsh** [comment]
Yeah, I believe this is intentional as `E203` is in conflict with PEP 8. (Ruff's implementation of `E203` does not flag this.)
